### PR TITLE
Fix doc generation

### PR DESCRIPTION
**Pull request by jlpedrosa**
Script was failing silently. see #1303 

CHanges:
* make the script fail on exit. (not that it matters, since helm-docs don't fail on error) return code is 0 on failures. :( 
* use go to run the image direclty instead of  downloading a version. Most of the CI images already have it.
* update the template so it's properly escaped.
* run the template (hence all the changes in the readme)

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
no, revert. the git command causes the fail if docs don't match .

**Comment by jlpedrosa** [comment]
> no, revert. the git command causes the fail if docs don't match .

I thought that was the goal to avoid having drifting config, that main don't match with actual docs...

**Review by viceice — COMMENTED** [review]
I'll work on a proper solution tomorrow.

we don't like to compile helm-docs from source,  it's much slower.

I'll take your required template changes into account.

**Comment by Lowaiz** [comment]
i also added fix a the template inside my MR about `set -e`, as I wanted to update the doc:
https://github.com/renovatebot/helm-charts/pull/1299/files#diff-d0ea23db559a530d067de34b08fee0a319e9564f62a80e373d31c813241aa392

File: charts/renovate/README.md [hunk 1]
```diff
@@ -118,11 +119,13 @@ can't write to the mounted PVC. For the current default user (`ubuntu`), the cor
 ## Renovate config templating
 
 Enable `renovate.configEnableHelmTpl` to use helm templates for generated renovate `config.json`.
-Allows you to reference values using `"{{ .Values.someValue }}"` in your config
+Allows you to reference values using `"` .Values.someValue `"` in your config
```
Comment (viceice) [inline charts/renovate/README.md:122]: this is wrong, the `{{` are required

File: charts/renovate/README.md [hunk 2]
```diff
@@ -118,11 +119,13 @@ can't write to the mounted PVC. For the current default user (`ubuntu`), the cor
...
 
 **NOTE**: setting `renovate.configEnableHelmTpl` to true means that you have to
-escape your config entries containing `{{` (i.e. `"key": "{{depName}}"`) in the
-value by wrapping it like: `"key": "{{ "{{depName}}" }}"`.
+escape your config entries containing {{ (i.e. `"key": "{{{depName}}"`) in the
+value by wrapping it like: `"key": "{{ "\{\{depName\}\}" \}\}"`.
+
+}}
```
Comment (viceice) [inline charts/renovate/README.md:128]: these are also wrong quotes

**Pull request closed without merging**